spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/publisher/PubSubPublisherTemplate.java

```diff
@@ -103,7 +105,8 @@ public void setMessageConverter(PubSubMessageConverter pubSubMessageConverter) {
 			@Override
 			public void onFailure(Throwable throwable) {
 				LOGGER.warn("Publishing to " + topic + " topic failed.", throwable);
-				settableFuture.setException(throwable);
+				PubSubDeliveryException pubSubDeliveryException = new PubSubDeliveryException(pubsubMessage, throwable.getMessage(), throwable);
```

Comment:
No, I don't want to see the same `throwable.getMessage()` here again on top of stack trace.
I can get it in the stack trace anyway parsing a `cause`.
What I want to see is an explicit message indicating that exactly here in the `PubSubTemplate` we couldn't delivered a message.

Reply:
+1
I think what you're logging should work here as well for the message:
`"Publishing to " + topic + " topic failed."`

Reply:
Updated the code.